diff --git a/cmake/modules/Libdispatch.cmake b/cmake/modules/Libdispatch.cmake
index 39b3e4bc531..9eeb26df01e 100644
--- a/cmake/modules/Libdispatch.cmake
+++ b/cmake/modules/Libdispatch.cmake
@@ -103,6 +103,8 @@ foreach(sdk ${DISPATCH_SDKS})
                           "-DCMAKE_ANDROID_NDK=${SWIFT_ANDROID_NDK_PATH}"
                           -DCMAKE_ANDROID_ARCH_ABI=${SWIFT_SDK_ANDROID_ARCH_${arch}_ABI}
                           -DCMAKE_ANDROID_API=${SWIFT_ANDROID_API_LEVEL}
+                          -DANDROID_PLATFORM=${ANDROID_PLATFORM}
+                          -DANDROID_ABI=${ANDROID_ABI}
                           -DBUILD_SHARED_LIBS=YES
                           -DENABLE_SWIFT=NO
                           -DBUILD_TESTING=NO
@@ -195,6 +197,8 @@ foreach(sdk ${DISPATCH_SDKS})
                             "-DCMAKE_ANDROID_NDK=${SWIFT_ANDROID_NDK_PATH}"
                             -DCMAKE_ANDROID_ARCH_ABI=${SWIFT_SDK_ANDROID_ARCH_${arch}_ABI}
                             -DCMAKE_ANDROID_API=${SWIFT_ANDROID_API_LEVEL}
+                            -DANDROID_PLATFORM=${ANDROID_PLATFORM}
+                            -DANDROID_ABI=${ANDROID_ABI}
                             -DBUILD_SHARED_LIBS=NO
                             -DENABLE_SWIFT=NO
                             -DBUILD_TESTING=NO
diff --git a/stdlib/cmake/modules/AddSwiftStdlib.cmake b/stdlib/cmake/modules/AddSwiftStdlib.cmake
index b41261feb2b..a33ccc3bfac 100644
--- a/stdlib/cmake/modules/AddSwiftStdlib.cmake
+++ b/stdlib/cmake/modules/AddSwiftStdlib.cmake
@@ -507,6 +507,9 @@ function(_add_target_variant_link_flags)
     endif()
   endif()
 
+  list(APPEND result "-Wl,-z")
+  list(APPEND result "-Wl,nostart-stop-gc")
+
   get_maccatalyst_build_flavor(maccatalyst_build_flavor
     "${LFLAGS_SDK}" "${LFLAGS_MACCATALYST_BUILD_FLAVOR}")
 
